Files/DataModels/SidebarPinnedModel.cs

```diff
@@ -96,7 +97,7 @@ public async void AddItem(string item)
 
         public async Task ShowHideRecycleBinItemAsync(bool show)
         {
-            await MainPage.SideBarItemsSemaphore.WaitAsync();
+            await SidebarControl.SideBarItemsSemaphore.WaitAsync();
```

Comment:
We shouldn't be referencing the sidebar control directly, technically, we should be able to spin up several sidebars and give each one their own list of items.

Reply:
@yaichenbaum But the `SideBarItemsSemaphore` was static and is static anyway. I just moved it there so it wasn't ugly placed in MainPage.

Reply:
Let's keep it in MainPageViewModel until we set it up properly since it will help keep the control decoupled.

Reply:
@yaichenbaum Can we just keep it where it is now? A quite a lot of files have been changed for this change and moving it to MainPageViewModel changes nothing.

Reply:
It means that one more thing would break if the sidebar control was removed, however, we can keep it where it is for now. Can you open an issue to track this?